File: android/src/main/java/com/ziggeo/utils/Events.java
package com.ziggeo.utils;

public final class Events {
    private Events() {
    }
    // Camera
    public static final String CAMERA_OPENED = "CameraOpened";
    public static final String CAMERA_CLOSED = "CameraClosed";

    // Common
    public static final String LOADED = "RecordingStarted";
    public static final String CANCELLED_BY_USER = "CancelledByUser";

    // Error
    public static final String ERROR = "Error";

    // Recorder
    public static final String MANUALLY_SUBMITTED = "ManuallySubmitted";
    public static final String RECORDING_STARTED = "RecordingStarted";
    public static final String RECORDING_STOPPED = "RecordingStopped";
    public static final String COUNTDOWN = "Countdown";
    public static final String RECORDING_PROGRESS = "RecordingProgress";
    public static final String READY_TO_RECORD = "ReadyToRecord";
    public static final String RERECORD = "Rerecord";

    // Streaming
    public static final String STREAMING_STARTED = "StreamingStarted";
    public static final String STREAMING_STOPPED = "StreamingStopped";

    // Camera hardware
    public static final String NO_CAMERA = "NoCamera";
    public static final String HAS_CAMERA = "HasCamera";

    // Mic hardware
    public static final String MIC_HEALTH = "MicrophoneHealth";
    public static final String NO_MIC = "NoMicrophone";
    public static final String HAS_MIC = "HasMicrophone";

    // Permissions
    public static final String ACCESS_GRANTED = "AccessGranted";
    public static final String ACCESS_FORBIDDEN = "AccessForbidden";

    // Uploader
    public static final String UPLOADING_STARTED = "UploadingStarted";
    public static final String UPLOAD_PROGRESS = "UploadProgress";
    public static final String VERIFIED = "Verified";
    public static final String PROCESSING = "Processing";
    public static final String PROCESSED = "Processed";
    public static final String UPLOADED = "Uploaded";

    // File selector
    public static final String UPLOAD_SELECTED = "UploadSelected";

    // Player
    public static final String PLAYING = "Playing";
    public static final String PAUSED = "Paused";
    public static final String ENDED = "Ended";
    public static final String SEEK = "Seek";
    public static final String READY_TO_PLAY = "ReadyToPlay";

    // QR scanner
    public static final String QR_DECODED = "QrDecoded";
}

File: android/src/main/java/com/ziggeo/utils/Events.kt
package com.ziggeo.utils

object Events {
    // Camera
    const val CAMERA_OPENED = "CameraOpened"
    const val CAMERA_CLOSED = "CameraClosed"

    // Common
    const val LOADED = "RecordingStarted"
    const val CANCELLED_BY_USER = "CancelledByUser"

    // Error
    const val ERROR = "Error"

    // Recorder
    const val MANUALLY_SUBMITTED = "ManuallySubmitted"
    const val RECORDING_STARTED = "RecordingStarted"
    const val RECORDING_STOPPED = "RecordingStopped"
    const val COUNTDOWN = "Countdown"
    const val RECORDING_PROGRESS = "RecordingProgress"
    const val READY_TO_RECORD = "ReadyToRecord"
    const val RERECORD = "Rerecord"

    // Streaming
    const val STREAMING_STARTED = "StreamingStarted"
    const val STREAMING_STOPPED = "StreamingStopped"

    // Camera hardware
    const val NO_CAMERA = "NoCamera"
    const val HAS_CAMERA = "HasCamera"

    // Mic hardware
    const val MIC_HEALTH = "MicrophoneHealth"
    const val NO_MIC = "NoMicrophone"
    const val HAS_MIC = "HasMicrophone"

    // Permissions
    const val ACCESS_GRANTED = "AccessGranted"
    const val ACCESS_FORBIDDEN = "AccessForbidden"

    // Uploader
    const val UPLOADING_STARTED = "UploadingStarted"
    const val UPLOAD_PROGRESS = "UploadProgress"
    const val VERIFIED = "Verified"
    const val PROCESSING = "Processing"
    const val PROCESSED = "Processed"
    const val UPLOADED = "Uploaded"

    // File selector
    const val UPLOAD_SELECTED = "UploadSelected"

    // Player
    const val PLAYING = "Playing"
    const val PAUSED = "Paused"
    const val ENDED = "Ended"
    const val SEEK = "Seek"
    const val READY_TO_PLAY = "ReadyToPlay"

    // QR scanner
    const val QR_DECODED = "QrDecoded"
}